src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax;
+
+public class Main {
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+        Figure[] figureArray = new Figure[6];
+        for (Figure figure : figureArray) {
+            figure = figureSupplier.getRandomFigure();
+            figure.draw();
+        }
```

Comment:
This loop does not actually populate the 'figureArray' with generated figures. The 'figure' variable is just a temporary reference and assigning a new object to it doesn't change the array content. Instead, you should use a traditional for loop with an index to populate the array.